### ISSUE TITLE
Support for Renovate Bot

### ISSUE THREAD

**Issue by miikeat** [jmfayard/refreshVersions#207]
Renovate Bot is only supporting updating version numbers in ```gradle.properties```. Probably it also doesn't understand the "```_```"-placeholder.

Is there a way to get Renovate Bot working while still using your Plugin with all it's advantages? I would love to have the feature of automatic PR creation for Dependency Updates but as we are using Bitbucket the number of possible applications to do this is quite low :(

**Comment by LouisCAD** [comment]
It's complicated and would certainly require work on Renovate bot's end. I'll keep this open, but this is low priority as I'm focused on version 1.0 (and on vacation).

**Comment by miikeat** [comment]
I'd be curious what's needed for this to work!
Maybe I'll find the time and could do the Renovate part...

**Comment by jmfayard** [comment]
I had a look at what Renovate Bot here https://github.com/galacticcouncil/substrate-api-explorer/issues/114 and here https://github.com/mulesoft/mule-migration-assistant/pull/406

It seems to me that it should be quite easy for the bot to run `$ ./gradle refershVersions` and then to read, analyse and write the file `versions.properties`. That's the whole point of it by the way, this file is machine readable and writable.

Given a `versions.properties` like this:
![](https://raw.githubusercontent.com/jmfayard/refreshVersions/master/docs/screenshots-usage/versions.properties_step01.png)
it could then 

```
- [ ] update version.kotlinx.coroutines to 1.3.7 from 1.3.1
```

And in the pull request it would just do 

```diff
## versions.properties
- version.kotlinx.coroutines=1.3.1
+ version.kotlinx.coroutines=1.3.7
```

Now ideally you would have a CI setup so that you run the unit tests for example to checck whether you have a compability problem with the new version.

(The only question would be how you filter the multiple options to just one)

**Comment by jmfayard** [comment]
Discover thanks to Twitter how renovabot currently works
https://github.com/renovatebot/renovate/blob/d858d85205cc99d7ecf617754612d2a17ff4eda0/lib/manager/gradle/gradle-updates-report.ts#L42

**Comment by vlsi** [comment]
AFAIK the approach here would be to add `gradle-refreshVersions` "manager" (like [gradle](https://github.com/renovatebot/renovate/tree/d858d85205cc99d7ecf617754612d2a17ff4eda0/lib/manager/gradle) and [gradle-wrapper](https://github.com/renovatebot/renovate/tree/d858d85205cc99d7ecf617754612d2a17ff4eda0/lib/manager/gradle-wrapper) )

**Comment by yschimke** [comment]
I came here to file this bug upon seeing https://github.com/renovatebot/renovate/pull/16684/files which adds support for kts files.  Does this still seem feasible?

**Comment by jmfayard** [comment]
@yschimke Kotlin scripts support in dependabot isn't related to refreshVersions.

Are you using refreshVersion already?

If yes there is no need to use dependabot.
I use a little GitHub Actions that runs or demand or according to a crontab,
which runs refreshVerisons, commit and open a pull request.
I can share it if you need it.

**Comment by vlsi** [comment]
@jmfayard , I believe the PR was just a sample integration.

> If yes there is no need to use dependabot

I'm afraid, that is the other way around.
Dependabot / Renovate helps to make it easier to weigh and test **different** dependencies. Dependabot/renovate generate human-readable PRs, they list changelog, etc, etc.

So it would make total sense to add something like PR 16684 above, except it should consume and update `versions.properties`.

**Comment by jmfayard** [comment]
If somoeone wants to work on dependabot for that, I'm totally fine.

It seems like they are doing manual parsing? That's a bit sad.

If they use Gradle there is nothing special they need to do query the list of dependencies and versions managed by refreshVersions.
And for updating `versions.properties` we can trivially add a task `./gradlew refreshVersionsUpdate --dependency "it.krzeminski:github-actions-kotlin-dsl" --version "0.22.0"`

**Comment by yschimke** [comment]
Yep, wasn't linking to that PR as a solution, more as a recent example showing they take contributions, and the complexity of this sort of support.

**Comment by jmfayard** [comment]
As I said, the complexity depend on whether they do manual parsing or can call Gradle.
If the later it should be pretty easy, for the former I wouldn't want to work on it personally.

**Comment by vlsi** [comment]
I do not think parsing `versions.properties` is harder than calling Gradle though.

**Comment by jmfayard** [comment]
With manual parsing, dependabot can't know that the key `version.kotest` is meant for `io.kotest:kotest-.*:*`

**Comment by yschimke** [comment]
nit: Is dependabot a generic term? I thought it and renovate were different products? Hopefully the solution could be adapted for both.

**Comment by vlsi** [comment]
>With manual parsing, dependabot can't know that the key version.kotest is meant for io.kotest:kotest-.*:*

Oh, I missed aliases.

**Comment by yschimke** [comment]
Yep.

Heres a renovate PR for it https://github.com/square/okhttp/pull/7386

**Comment by jmfayard** [comment]
Ah the issue is resolved then.
Then it works with refreshVersions too, we don't do anything special.

**Comment by simonbasle** [comment]
> nit: Is dependabot a generic term? I thought it and renovate were different products? Hopefully the solution could be adapted for both.

definitely not a generic term @yschimke. Dependabot and Renovate are two competitors (Dependabot doesn't indeed support version catalog Gradle feature).

also, did you link to the wrong PR in your last comment? I don't understand the relation with the current discussion (this is just an example of a Renovate PR) and I don't understand why @jmfayard comes to the conclusion the issue is resolved 🤔

**Comment by yschimke** [comment]
Yep, I confused everyone with a PR showing support for kts. Sorry for confusing this message.

> I came here to file this bug upon seeing https://github.com/renovatebot/renovate/pull/16684/files which adds support for kts files.
